src/box/memtx_tree.cc

```diff
@@ -1426,7 +1426,8 @@ memtx_tree_index_create_iterator(struct index *base, enum iterator_type type,
 	}
 	iterator_create(&it->base, base);
 	it->pool = &memtx->iterator_pool;
-	it->base.next = tree_iterator_start<USE_HINT>;
+	it->base.next_raw = tree_iterator_start_raw<USE_HINT>;
+	it->base.next = memtx_iterator_next;
```

Comment:
Here and in similar places you can simply use `it->base.next = it->base.next_raw` in case if there is no compression.